### ISSUE TITLE
Some suggestions

### ISSUE THREAD

**Issue by zwz** [massifrg/prosemirror-tables-sections#1]
I like this module. I tried the demo, and here are some suggestions:
1. set column width with percentage value. AFAIK, pandoc only treats `<col>` with percentage width value. And percentage value is good for fitting to various-size screens. So I would suggest set `data-colwidth` of `td` to percentage value as well (`data-colwidth="80"` means it takes 80% of width)
BTW: I cannot adjust the size in Firefox. This exists for a long time from prosemirror-tables.

2. improve cursor movement. If I move the cursor with up/down arrow key in the demo, it will skip table caption(both keys in firefox, only up key in chrome). And if I move the cursor out of table caption by right arrow key in firefox, it jumps out of the table. And if you hit tab key at the end of `thead/tbody`, it will select the next whole `tbody`, which I think is not expected. 

3. customizable schema. I would like to customize the content of the table, like a **required** table caption instead of an optional one.

**Comment by massifrg** [comment]
Thank you for your feedback.
Column widths are the last thing I managed to set (almost) right, because the way prosemirror-tables-sections creates the colgroup element is different (through Decorations) from prosemirror-tables (a custom EditorView).
I preferred publishing it earlier to get some feedback :)
I agree that percentages are more useful.

I'll look at cursor movement.

You can already define a different schema with a required caption.
prosemirror-tables defines table roles (the tableRole field in the schema), so that you can change the names of the nodes;
you can call the table node of your schema "myTable" instead of "table", but it will behave as a table as long as its tableRole is "table". The other roles are "row" (`tr`), "cell" (`td`), "header_cell" (`th`). I've added the "table_caption" (`caption`) and "table_section" (`thead`, `tbody` and `tfoot`) roles.

**Comment by zwz** [comment]
Yes, if `colgroup` is inherent in the schema, then there is no need to keep the `data-colwidth` for each cell.

Anyway, thank your for your work.

**Comment by massifrg** [comment]
Cursor and tab keys should work better now.
I still don't like the behavior of SHIFT+cursor when started inside the caption, but I'll leave it for future improvements.
Column resizing is not perfect but workable.

**Comment by zwz** [comment]
Yes, it is better now!

I cannot test column resizing in firefox, it is not available for a long time from prosemirror-tables.

**Comment by zwz** [comment]
Yes, it is better now!

I cannot test column resizing in firefox, it is not available for a long time from prosemirror-tables.

**Comment by massifrg** [comment]
I'm using firefox 102.9.0esr (64 bit) in Debian GNU/Linux 11 and column resizing works.
You could try firefox in a different environment (another user) or disable extensions selectively.
